src/main/java/com/rocketden/main/service/SubmitService.java

```diff
@@ -58,4 +105,58 @@ public SubmissionDto submitSolution(Game game, SubmissionRequest request) {
 
         return GameMapper.submissionToDto(submission);
     }
+
+    // Sends a POST request to the tester service to judge the user submission
+    protected Submission callTesterService(TesterRequest request) {
+        try {
+            HttpPost post = new HttpPost(getTesterUrl());
+
+            StringEntity stringEntity = new StringEntity(gson.toJson(request));
+            post.setEntity(stringEntity);
+            post.setHeader("Content-type", "application/json");
+
+            HttpResponse response = httpClient.execute(post);
+            String jsonResponse = EntityUtils.toString(response.getEntity());
+
+            TesterResponse testerResponse = getResponseFromJson(jsonResponse);
+
+            // TODO: logic to convert TesterResponse into Submission object
+            Submission submission = new Submission();
+            submission.setNumCorrect(request.getProblem().getTestCases().size());
+            submission.setNumTestCases(request.getProblem().getTestCases().size());
+            submission.setPlayerCode(new PlayerCode(request.getCode(), request.getLanguage()));
+
+            return submission;
+        } catch (Exception e) {
+            // If in debug mode (tester is unavailable), return a dummy submission
+            if (getDebugMode()) {
+                return getDummySubmission(request);
+            }
+
+            throw new ApiException(GameError.TESTER_ERROR);
+        }
+    }
+
+    // Is null in certain testing environments; if so, return a default value
+    private String getTesterUrl() {
+        if (testerUrl == null) {
+            return "http://localhost:8080";
+        }
+        return testerUrl;
+    }
+
+    // Is null in certain testing environments; if so, return true by default
```

Comment:
When is this value null?

Reply:
Only when running tests that have mocked the submit service class (for now that's just SubmitServiceTests I think)